lib/controllers/room_chat_controller.dart

```
@@ -44,6 +45,27 @@ class RoomChatController extends GetxController {
     log(appwriteUpcommingRoom.toString());
   }
 
+  // delete method
+  Future<void> deleteMessage(String messageId) async {
+    try {
+      Message messageToDelete = messages.firstWhere(
+        (msg) => msg.messageId == messageId,
+      );
+      messageToDelete = messageToDelete.copyWith(content: '', isDeleted: true);
+
+      await databases.updateDocument(
+        databaseId: masterDatabaseId,
+        collectionId: chatMessagesCollectionId,
+        documentId: messageId,
+        data: messageToDelete.toJsonForUpload(),
+      );
+      log('Message deleted successfully');
+    } catch (e) {
+      log('Error deleting message: $e');
```

Instead of silent failures it would be better to show the user that deletion has failed. To keep the controller decoupled you can rethrow the exception from here and then catch it on the screen